### PR TITLE
Update `tpu_cores` flag with accelerator and devices flag

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Update `tpu_cores` flag with accelerator and devices flag

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @kaushikb11 @rohitgr7 @justusschock @awaelchli

### CODE REVIEW

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: tests/models/test_tpu.py [hunk 1]
```diff
@@ -319,9 +319,9 @@ def test_broadcast(rank):
 def test_tpu_choice(tmpdir, tpu_cores, expected_tpu_id, error_expected):
     if error_expected:
         with pytest.raises(MisconfigurationException, match=r".*tpu_cores` can only be 1, 8 or [<1-8>]*"):
-            Trainer(default_root_dir=tmpdir, accelerator="tpu", devices=tpu_cores)
+            Trainer(default_root_dir=tmpdir, tpu_cores=tpu_cores)
```
Comment (rohitgr7) [inline tests/models/test_tpu.py:322]: qq: this isn't completely compatible yet?

**Review by ananthsub — APPROVED** [review]

**Review by four4fish — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: tests/models/test_tpu.py (hunk 1 above)
Comment (kaushikb11) [reply tests/models/test_tpu.py]: Updated the test!

**Review by tchaton — APPROVED** [review]
LGTM !
